File: 372 - Android/Ashman/app/src/main/java/com/benedetto/lars/ashman/enums/GhostColor.java
package com.benedetto.lars.ashman.enums;

import java.util.Random;

public enum GhostColor {
    RED, ORANGE, PINK, BLUE,;

    public static GhostColor getRandomColor() {
        Random random = new Random();
        int rand = random.nextInt(4);
        switch (rand) {
            case 0:
                return RED;
            case 1:
                return ORANGE;
            case 2:
                return PINK;
            case 3:
                return BLUE;
        }
        return BLUE;
    }
}

File: 372 - Android/Ashman/app/src/main/java/com/benedetto/lars/ashman/enums/GhostColor.kt
package com.benedetto.lars.ashman.enums

import java.util.Random

enum class GhostColor {
    RED, ORANGE, PINK, BLUE;

    companion object {

        val randomColor: GhostColor
            get() {
                val random = Random()
                val rand = random.nextInt(4)
                return when (rand) {
                    0 -> RED
                    1 -> ORANGE
                    2 -> PINK
                    else -> BLUE
                }
            }
    }
}
